settings.nix

```diff
@@ -0,0 +1,50 @@
+# `./settings.nix` contains project wide settings such as
+#   
+#   - `perSystem.settings.devShellTools`: packages to include in every developer shell
+#   e.g. pre-commit tools
+# 
+#   - `perSystem.settings.devShellHook`: shell script to invoke in all devShells (e.g.
+#   to ensure that pre-commit hooks are installed properly)
+#
+{ flake-parts-lib, lib, ... }:
+{
+
+  options = {
+    perSystem = flake-parts-lib.mkPerSystemOption
+      ({ config, ... }:
+        {
+          options = {
+            settings = {
+              devShellTools = lib.mkOption {
+                type = lib.types.listOf lib.types.package;
+                description = ''
+                  Tools to be included in all devshells.
+                '';
+              };
+              devShellHook = lib.mkOption {
+                type = lib.types.str;
+                description = ''
+                  Shell script to be invoked in all devshells.
+                '';
+              };
+            };
+          };
+
+          config = {
+            settings.devShellTools =
+              [
+                # Note(jaredponn): Include other useful developer devShellTools
+                # that should be available for _all_ dev shells here
+              ];
```

Comment:
@gnumonik @Luis-omega if there's any developer tools we find helpful to always have, feel free to toss them here.